Backend/CarRentalApp/CarRentalApp/Models/DAOs/RefreshTokenDAO.cs

```diff
@@ -0,0 +1,45 @@
+using CarRentalApp.Contexts;
+using CarRentalApp.Models.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace CarRentalApp.Models.DAOs
+{
+    public class RefreshTokenDAO
+    {
+        private readonly CarRentalDbContext _carRentalDbContext;
+
+        public RefreshTokenDAO(CarRentalDbContext carRentalDbContext)
+        {
+            _carRentalDbContext = carRentalDbContext;
+        }
+
+        public Task InsertAsync(RefreshToken refreshToken)
+        {
+            _carRentalDbContext.RefreshTokens.Add(refreshToken);
+            return _carRentalDbContext.SaveChangesAsync();
+        }
+        
+        public async Task<RefreshToken?> GetByTokenStringAsync(string refreshTokenString)
+        {
+            return await _carRentalDbContext.RefreshTokens
+                .FirstOrDefaultAsync(t => t.Token == refreshTokenString);
+        }
+
+        public Task DeleteAsync(RefreshToken refreshToken)
+        {
+            _carRentalDbContext.RefreshTokens.Remove(refreshToken);
+            return _carRentalDbContext.SaveChangesAsync();
+        }
+
+        public async Task DeleteRelatedTokensByUserIdAsync(Guid userId)
```

Comment:
ByUserId(iserId) already shows that this token is related to user
```suggestion
        public async Task DeleteTokensByUserIdAsync(Guid userId)
```